### ISSUE TITLE
addon-docs: Allow setting `argTypes` for `subcomponents`

### ISSUE THREAD

**Issue by randak** [storybookjs/storybook#14710]
**Is your feature request related to a problem? Please describe**
I have a situation where I want to use `subcomponents` but I need to be able to customize some of the values displayed in the generated table. As far as I can tell, I can only apply `argTypes` customization to the main `component`, not the `subcomponent`. 

**Describe the solution you'd like**
A method of setting `argTypes` for subcomponents. I'm not sure what the best config structure would be for this, and am open to suggestions.

**Describe alternatives you've considered**
N/A

**Are you able to assist to bring the feature to reality?**
Yes, with pointers on where the relevant code is.

**Additional context**
N/A

**Comment by AmirTugi** [comment]
How about 
```tsx
export default {
  title: 'Components/Calendar',
  component: Calendar,
  subcomponents: { 
    CalendarLegend: {
      component: CalendarLegend,
      argTypes: {
        label: { control: 'input' },
      },
    },
  },
  argTypes: {
    onSelect: { action: 'selected' },
  }
};
```

And in case we don't want to add manual `argTypes` it will be left as in today - 
```tsx
export default {
  title: 'Components/Calendar',
  component: Calendar,
  subcomponents: { CalendarLegend },
  argTypes: {
    onSelect: { action: 'selected' },
  }
};
```

**Comment by Fernker** [comment]
Anyone know of a workaround for this?  I find it incredible that `argTables` is considered a first-class citizen but there's no support for it on subcomponents.

**Comment by curtvict** [comment]
@AmirTugi to be clear, this has yet to be added as a feature, correct?

**Comment by vhoyer** [comment]
> @AmirTugi to be clear, this has yet to be added as a feature, correct?

correct.

**Comment by vhoyer** [comment]
> 

I like this syntax, this was the first one I tried to see if it worked on my project before being greeted by a lovely message saying what was the expected format for the `subcomponents` key.

The only problem I have with this is that, since this is an open format, this proposal would not be "ignorable" by other tools working with this format, since it's technically a breaking change, I'd like to propose another API for getting around this problem:

```js
// proposal "subcomponent in argTypes"
export default {
  title: 'Components/Calendar',
  component: Calendar,
  subcomponents: { 
    CalendarLegend,
  },
  argTypes: {
    onSelect: { action: 'selected' },
    CalendarLegend: {
      label: { control: 'input' },
    },
  },
};
```

This could easily be ignored by other tools applying the CSF, and it gets the job done, I don't suppose we'd have the need to configure anything else in the subcomponents other than the `ArgTable` anyway, right?

Another possibility I just came up with while typing this message was this other API, but I preffer the "subcomponent in argTypes" proposal:

```js
// proposal "subcomponent as namespace for argTypes"
export default {
  title: 'Components/Calendar',
  component: Calendar,
  subcomponents: { 
    CalendarLegend,
  },
  argTypes: {
    'onSelect': {
      action: 'selected',
    },
    'CalendarLegend/label': {
      control: 'input',
    },
  },
};
```

**Comment by mtirimie1** [comment]
Any progress on this? Running into quite a few scenarios on my current project where this would alleviate a lot of confusion for the end user.

**Comment by vhoyer** [comment]
@shilman, since we removed the "needs triage" does it mean the its green lit to be implemented? do you have any considerations about the proposed APIs? Which do you think should be implemented?

**Comment by shilman** [comment]
@vhoyer No, it just means that I don't want it to show up anymore when I look through the "needs triage" issues in Storybook. I'll let @JReinhold comment on the future of subcomponents...

**Comment by JReinhold** [comment]
We're deprecating `subcomponents` in 7.0. The API is very limiting and hard to improve in (as evident by this issue and others). It is somewhat of a half-baked feature, that we want to rethink and do better.

Subcomponents will be removed from the default DocsPage in 7.0, but since it's still available in the now deprecated `<ArgsTable />` doc block, they can be re-added by defining your own DocsPage globally.

We don't have a clear path forward from this yet, we want to collect feedback from the community about what they want from their subcomponent documentation and how they want to use it.

**Comment by JReinhold** [comment]
We've since reverted the decision to remove subcomponents support, so this feature request is valid again.

See https://github.com/storybookjs/storybook/issues/20782#issuecomment-1892178858

**Comment by AndreiRupertti** [comment]
Any progress on this ?

**Comment by noxickon** [comment]
hey, I want this feature too, and hopefully this can be a solution for how to implement it

you already support this:
```js
  align: {
    control: 'select',
    options: ['left', 'right'],
    description:
      'Item: Content alignment relative to central icon (left = content left of icon, right = content right of icon)',
    table: { category: 'Timeline.Item' },
  },
```
Result: 
<img width="590" height="199" alt="Image" src="https://github.com/user-attachments/assets/5f4ae1f0-73f3-44f5-b12b-ad310d8eee35" />

instead of category, why not use a tab key? or something like this? maybe this approach has fewer breaking changes (hopefully)?

```js
  align: {
    control: 'select',
    options: ['left', 'right'],
    description:
      'Item: Content alignment relative to central icon (left = content left of icon, right = content right of icon)',
    table: { tab: 'Timeline.Item' },
  },
```

**Comment by sglza** [comment]
Would also love to see this added :)
